drivers/hid/hidparse/hidp_km.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * PROJECT:     ReactOS HID Parser Library
+ * LICENSE:     GPL-3.0-or-later (https://spdx.org/licenses/GPL-3.0-or-later)
+ * FILE:        drivers/hid/hidparse/hidp_km.c
+ * PURPOSE:     HID Parser kernel mode
+ * COPYRIGHT:   Copyright (C) Michael Martin (michael.martin@reactos.org)
+ *              Copyright (C) Johannes Anderwald (johannes.anderwald@reactos.org)
+ *              Copyright (C) Roman Masanin (36927roma@gmail.com) 2022
+ */
+
+#include "parser.h"
+#include <hidparser.h>
+#include <hidpmem.h>
+
+#define NDEBUG
+#include <debug.h>
+
+NTSTATUS
+NTAPI
+HidP_GetCollectionDescription(
+    IN PHIDP_REPORT_DESCRIPTOR ReportDesc,
```

Comment:
For new functions, please use SAL2 annotations (`_In_`, `_Out_`, etc.)

Reply:
ok let's proceed with the annotations later then (and separate commit? 🤔)

Reply:
yep, it can take some time, so making it in separate commit sounds like a good idea.